Lib/test/test_asyncio/test_events.py

```diff
@@ -670,6 +670,17 @@ def test_create_connection_local_addr(self):
             self.assertEqual(port, expected)
             tr.close()
 
+    def test_create_connection_local_addr_1(self):
+        # See https://github.com/python/cpython/issues/86508
+        port1 = socket_helper.find_unused_port()
+        port2 = socket_helper.find_unused_port()
+        f = self.loop.create_connection(
+            lambda: MyProto(loop=self.loop),
+            'localhost', port1, local_addr=('localhost', port2))
```

Comment:
Relying on `localhost` doesn't seem very robust as a test -- who knows whether this reproduces the situation or not (it may be only IPv4 or only IPv6, rather than mixed; only the latter will reproduce the bug).

Reply:
Fixed.